graph/tarjan.cpp

```diff
@@ -0,0 +1,269 @@
+            weight,
+            std::make_pair(uvertex, vvertex)));  ///< adding edge to the vector
+
+        isDirected=true;
+        g[uvertex].emplace_back(std::make_pair(vvertex, weight)); ///< adding edge to adjcency list
+    }
+
+
+    /**
+     * @brief returns a vector containing list of nodes, in order in which they are encountered in BFS call
+     * @param start_vertex formal argument that represents the starting vertex for BFS
+     * @returns vector<int>
+    */
+    std::vector<int> BFS(int start_vertex) {
+        std::vector<int> ans;  ///< This vector will hold the BFS result
+        std::vector<bool> visited(vertex,false);  ///< This vector will keep track of visited vertices
+        std::queue<int> q;  ///< This queue will keep track of vertices to visit next
+        q.push(start_vertex);  ///< Add the starting vertex to the queue
+        visited[start_vertex] = true;  ///< Mark the starting vertex as visited in the visited vector
+
+        while (!q.empty())  ///< As long as there are vertices left to explore
+        {
+            int u = q.front();    ///< Get the vertex from the front of the queue
+            ans.emplace_back(u);  ///< Add it to the BFS result vector
+            q.pop();              ///< Remove it from the queue
+            for (auto i : g[u])  ///< Explore all vertices adjacent to the current vertex
+            {
+                if (visited[i.first] ==  false)  ///< If the vertex has not been visited before
+                {
+                    q.push(i.first);  ///< Add it to the queue
+                    visited[i.first] = true;  ///< Mark it as visited in the visited vector
+                }
+            }
+        }
+        return ans;  ///< Return the BFS result vector
+    }
+
+    /**
+     * @brief This function implements a Depth First Search (DFS) algorithm.
+     * @param start_vertex represents the starting vertex for dfs call
+     * @param result it will contain the list of nodes that are visited in a dfs call
+     * @returns void
+     */
+
+    void DFS(int start_vertex, std::vector<int> &result) {
+        static std::vector<bool> visited(vertex, false);  ///< This vector will keep track of visited vertices
+        visited[start_vertex] = true;  ///< Mark the starting vertex as visited
+        result.emplace_back(start_vertex);  ///< Add the starting vertex to the DFS result vector
+
+        for (auto i : g[start_vertex])  ///< Explore all vertices adjacent to the current vertex
+        {
+            if (!visited[i.first])  ///< If the vertex has not been visited before
+            {
+                DFS(i.first,result);  ///< Recursively call DFS on the unvisited vertex
+            }
+        }
+    }
+};
+
+}
+
+// // This function implements Tarjan's algorithm to find articulation points in
+// an undirected graph represented using an adjacency list
+
+void tarjans_ap(graph::Graph &tg, int start_vertex, std::set<int> &result) {
```

Comment:
1. I believe this function must be inside the `graph` namespace for consistency and organization.
2. The function's missing proper Doxygen documentation (check out our [contributing guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md) for more information).
3. Wrong indentation. The comments should be adjusted as well for better readability.

Let us know if you need any help with any of these changes. 🙂